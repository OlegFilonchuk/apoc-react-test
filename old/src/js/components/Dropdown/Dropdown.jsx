import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { isUndefined } from 'lodash';
import './Dropdown.less';
import DropdownOptionLink from './DropdownOptionLink';
import DropdownOption from './DropdownOption';

const keyCodeToIndexModifier = {
  13: 0, // key enter
  38: -1, // key arrow up
  40: 1 // key arrow down
};

export default class Dropdown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      currentOption: {},
      useButtonText: true,
      search: ''
    };

    const children = React.Children.toArray(this.props.children);

    if (children) {
      this.setSelectedOption(children);
    }

    this.closeOnOutsideClick = this.closeOnOutsideClick.bind(this);
  }

  componentWillMount() {
    window.document.addEventListener('click', this.closeOnOutsideClick);
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.children || nextProps.children.length === 0) {
      return;
    }

    const selectedOption = nextProps.children.find(option => option.props.selected);
    const defaultSelectedOption = nextProps.children[0];
    const useButtonText = !selectedOption && nextProps.recomputeUseButton ? true : this.state.useButtonText;

    this.setState(prevState => ({
      ...prevState,
      currentOption: (selectedOption || defaultSelectedOption).props,
      useButtonText
    }));
  }

  componentWillUnmount() {
    window.document.removeEventListener('click', this.closeOnOutsideClick);
  }

  onChange = () => {
    if (this.props.button.props.inputType === 'input' && !this.state.isOpen) {
      this.setState({ isOpen: true });
    }
  };

  getOptions() {
    const filtered = this.props.filterable ? this.dropdownOptions : this.props.children;

    return React.Children.map(filtered, child => React.cloneElement(child, this.getChildProps(child)));
  }

  get anyOptionSelected() {
    return Array.isArray(this.props.children) && this.props.children.some(child => child.props.selected);
  }

  get shouldUseTextOnButton() {
    return (
      !this.anyOptionSelected &&
      ((this.state.useButtonText && this.props.button.props.children) || this.optionType !== DropdownOption)
    );
  }

  get buttonText() {
    if (this.shouldUseTextOnButton) {
      return this.props.button.props.children;
    }

    return this.state.currentOption.label;
  }

  getButton() {
    return React.cloneElement(this.props.button, {
      onClick: (...args) => {
        if (this.props.button.props.onClick) {
          this.props.button.props.onClick(...args);
        }

        this.toggle(...args);
      },
      onKeyDown: (...args) => {
        if (this.props.button.props.onKeyDown) {
          this.props.button.props.onKeyDown(...args);
        }
        this.changeCurrentOptionOnKeyPressed(...args);
      },
      onChange: (...args) => {
        if (this.props.button.props.onChange) {
          this.props.button.props.onChange(...args);
        }
        this.onChange(...args);
      },
      isExpanded: this.state.isOpen,
      children: this.props.button.props.inputType === 'searchDropdown' ? undefined : this.buttonText
    });
  }

  /**
   * @param currentOption {DropdownOptionProps}
   */
  setCurrentOption(currentOption) {
    if (currentOption) {
      this.setState(prevState => ({ ...prevState, currentOption, useButtonText: false }));
    }
  }

  setSelectedOption(children) {
    if (!children[0]) {
      return;
    }

    this.optionType = children[0].type;
    if (this.optionType === DropdownOption) {
      const selectedOption = children.find(option => option.props.selected);

      this.state.currentOption = selectedOption ? selectedOption.props : children[0].props;
      this.state.useButtonText = !selectedOption;
    }
  }

  get filteredList() {
    const search = this.state.search.toLowerCase();

    return this.props.children.filter(
      item => typeof item.props.label !== 'string' || item.props.label.toLowerCase().includes(search)
    );
  }

  get dropdownOptions() {
    const result = this.filteredList;

    return result.length === 0 ? [<DropdownOption value="empty" label={this.props.emptyMessage} disabled />] : result;
  }

  get suffix() {
    const {
      withSuffix: { suffixWrapperClassName, suffix }
    } = this.props;

    return suffix ? <div className={suffixWrapperClassName}>{suffix}</div> : null;
  }

  getChildProps(child) {
    if (this.optionType === DropdownOptionLink) {
      return {
        onClick: (...args) => {
          if (child.props.onClick) {
            child.props.onClick(...args);
          }
          if (child.type === this.optionType && !child.props.disabled) {
            this.close();
          }
        }
      };
    }

    // DropdownOption
    return {
      onClick: (...args) => {
        if (child.props.onClick) {
          child.props.onClick(...args);
        }
        this.setState(prevState => ({ ...prevState, search: '' }));
        this.close();

        if (!isUndefined(child.props.value)) {
          this.setCurrentOption(child.props);
          this.props.onChange(child.props);
        }
      },
      selected: this.state.currentOption.value === child.props.value,
      name: this.props.name
    };
  }

  changeCurrentOptionOnKeyPressed(e) {
    if (!this.state.isOpen || !this.props.useSubmitOnEnter) {
      return;
    }

    const indexModifier = keyCodeToIndexModifier[e.keyCode];

    if (indexModifier === undefined) {
      return;
    }
    e.preventDefault();

    /**
     * @type {DropdownOptionProps[]}
     */
    const options = React.Children.map(this.props.children, child => child.props) || [];
    const optionProps = options.filter(childProps => !childProps.disabled);
    const currentIndex = optionProps.findIndex(option => option.value === this.state.currentOption.value);
    const newIndex = Math.min(Math.max(currentIndex + indexModifier, 0), optionProps.length - 1);

    this.setCurrentOption(optionProps[newIndex]);
    this.submitOnEnterPress(e);
  }

  closeOnOutsideClick(e) {
    if (!this.state.isOpen) {
      return;
    }

    if (this.root === e.target || !this.root.contains(e.target)) {
      this.close();
    }
  }

  submitOnEnterPress = event => {
    if (
      this.props.useSubmitOnEnter &&
      event.key === 'Enter' &&
      React.Children.toArray(this.props.children).length > 0
    ) {
      this.props.onChange(this.state.currentOption);
      this.close();
    }
  };

  toggle(e, isOpen) {
    const toggleState = isOpen !== undefined ? isOpen : !this.state.isOpen;

    this.setState(prevState => ({ ...prevState, isOpen: toggleState, search: '' }));
  }

  close() {
    this.setState(prevState => ({ ...prevState, isOpen: false }));
  }

  handleSearchInput = e => {
    const search = e.target.value;

    this.setState(prevState => ({ ...prevState, search }));
  };

  render() {
    const {
      className,
      fullWidth,
      filterable,
      button: {
        props: { id }
      }
    } = this.props;

    const mainDivClassName = classNames(
      'sc-dropdown',
      {
        'sc-open': this.state.isOpen,
        'sc-dropdown--fullwidth': fullWidth
      },
      className
    );

    return (
      /* eslint-disable no-return-assign */
      <div className={mainDivClassName} ref={ref => (this.root = ref)}>
        {this.getButton()}
        <ul className="sc-dropdown-menus" aria-labelledby={id}>
          {filterable ? (
            <div className="dropdown-filter">
              <input className="dropdown-filter__input" value={this.state.search} onChange={this.handleSearchInput} />
              <i className="sc-icon-magnifier dropdown-filter__icon" />
            </div>
          ) : null}
          {this.getOptions()}
        </ul>
        {this.suffix}
      </div>
      /* eslint-enable no-return-assign */
    );
  }
}

Dropdown.propTypes = {
  /**
   * Called only when [DropdownOption](#dropdownoption) are used. First argument are props of selected
   * [DropdownOption](#dropdownoption).
   */
  onChange: PropTypes.func.isRequired,

  /**
   * Button to toggle the dropdown. Should be instance of [DropdownButton](#dropdownbutton) or any other component
   * that shares same props API.
   */
  button: PropTypes.element.isRequired,

  /**
   * Necessary only when [DropdownOption](#dropdownoption) is used. Sets a name every option's `input`.
   */
  name: (props, ...args) => {
    const stringError = PropTypes.string(props, ...args);

    if (stringError) {
      return stringError;
    }

    if (!props.children) {
      return null;
    }

    if (!props.name) {
      const children = React.Children.toArray(props.children);
      const optionType = children[0].type;

      if (optionType === DropdownOption) {
        return new Error('Prop `name` cannot be empty if options are of <DropdownOption> type');
      }
    }

    return null;
  },

  /**
   * Css classes of container.
   */
  className: PropTypes.string,

  /**
   * Render the dropdown full width
   */
  fullWidth: PropTypes.bool,

  /**
   * Enable/disable submit on enter press
   */
  useSubmitOnEnter: PropTypes.bool,

  /**
   * Allows to filter [DropdownOption](#dropdownoption) options using given phrase
   */
  filterable: PropTypes.bool,

  /**
   * Content of empty filter results message
   */
  emptyMessage: PropTypes.string,

  /**
   * Defines options of dropdown. Only [DropdownOption](#dropdownoption) or [DropdownOptionLink](#dropdownoptionlink)
   * can be used. [DropdownOption](#dropdownoption) non-strings labels are always visible while filtering.
   */
  children: props => {
    if (!props.children) {
      return null;
    }

    const children = React.Children.toArray(props.children);
    const optionType = children[0].type;
    const hasFirstChildCorrectType = optionType === DropdownOptionLink || optionType === DropdownOption;

    if (!hasFirstChildCorrectType) {
      return new Error(`
        All <Dropdown/> children must be instances of either one <DropdownOptionLink/> or <DropdownOption/>
    `);
    }

    if (optionType === DropdownOption) {
      const uniqueValueProps = children
        .map(child => child.props.value)
        .filter((childValue, i, childrenValues) => childrenValues.indexOf(childValue) === i);

      if (uniqueValueProps.length !== children.length) {
        return new Error('<DropdownOption>s do not have unique values');
      }
    }

    if (props.filterable && optionType === DropdownOption) {
      if (children.some(child => !!child.props.label.type)) {
        return new Error('<DropdownOption>s non-strings labels are always visible while filtering');
      }
    }

    return children.reduce(error => {
      if (error) {
        return error;
      }

      return null;
    }, null);
  },

  /**
   * Allows to set up suffix near dropdown
   */
  withSuffix: PropTypes.shape({
    suffix: PropTypes.string,
    suffixWrapperClassName: PropTypes.string
  }),

  /**
   * Indicate to recompute `useButtonText` value once props will change or maintain internal value only
   */
  recomputeUseButton: PropTypes.bool
};

Dropdown.defaultProps = {
  name: '',
  className: '',
  filterable: false,
  recomputeUseButton: false,
  emptyMessage: 'No Filtered Data',
  children: [],
  fullWidth: false,
  withSuffix: {},
  useSubmitOnEnter: false
};
