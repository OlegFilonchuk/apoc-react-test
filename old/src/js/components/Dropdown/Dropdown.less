@import '~chameleon/less/variables.less';

@disabled-text-color: #B2B2B2;

.dropdown-button-container {
  display: inline-block;
}

.dropdown-option-disabled {
  pointer-events: none;
  opacity: 0.5;
}

.sc-dropdown.sc-dropdown--fullwidth {
  margin-right: 10px;

  .sc-dropdown-menus {
    width: 100%;
    left: 0;
  }

  .sc-btn.sc-btn-dropdown {
    font-family: @font-semibold;

    width: 100%;
    position: relative;
    text-align: left;
    margin: 0;

    .sc-icon-down-arrow {
      left: auto;
      right: 10px;
      position: absolute;
      top: 10px;
    }
  }
}

// DropdownOptionLink icon related styles
.dropdown-option,
.dropdown-option-link {
  .with-icon {
    padding-left: 5px;

    .icon {
      padding: 0 5px;
    }
  }
}

li.dropdown-option-link.disabled a {
  color: @disabled-text-color;
  cursor: not-allowed;
}

.dropdown-filter {
  position: relative;
  padding: (@padding-sm-vertical) (@padding-sm-horizontal * 2);

  &__input {
    width: 100%;
    padding-right: (@padding-base-horizontal * 3);
  }

  .dropdown-filter__icon {
    position: absolute;
    top: @padding-sm-vertical;
    right: (@padding-sm-horizontal * 3);
    font-size: @font-size-xl;
    color: @primary;
  }
}
