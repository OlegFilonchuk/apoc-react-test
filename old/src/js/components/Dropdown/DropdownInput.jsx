import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

export default class DropdownInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: this.props.value
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    const value = e.target.value;

    this.setState({ value }, () => this.props.onChange(value));
  }

  render() {
    const { className, onClick, isExpanded, inputType, placeholder } = this.props;
    const expanded = !(this.root !== document.activeElement && isExpanded);

    return (
      <input
        ref={ref => {
          this.root = ref;
        }}
        className={`sc-btn ${className}`}
        type={inputType}
        onClick={e => onClick(e, expanded)}
        onChange={this.handleChange}
        onKeyDown={this.props.onKeyDown}
        value={this.state.value}
        placeholder={placeholder}
        {...getDataTestElementProps(this.props)}
      />
    );
  }
}

DropdownInput.propTypes = {
  onKeyDown: PropTypes.func,
  onClick: PropTypes.func,
  isExpanded: PropTypes.bool,
  className: PropTypes.string,
  inputType: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  ...dataTestElementPropTypes
};

DropdownInput.defaultProps = {
  className: 'sc-btn-default',
  value: '',
  isExpanded: false,
  inputType: 'input',
  onKeyDown: () => true,
  onClick: () => true,
  onChange: () => true,
  placeholder: ''
};
