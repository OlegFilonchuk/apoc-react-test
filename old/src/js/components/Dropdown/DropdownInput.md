Purpose of this component is to be used together with [Dropdown](#dropdown).

## Standalone example

    <DropdownInput onChange={(e) => { console.log('Value changed', e); }} />

