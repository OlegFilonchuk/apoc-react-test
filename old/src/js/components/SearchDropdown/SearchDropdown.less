@search-box-background: #fff;
@search-box-border-color: #0290d7;
@search-box-placeholder-color: #9a9a9a;

.search-dropdown {
  position: relative;

  display: inline-block;

  &--icon {
    position: absolute;
    top: 5px;
    right: 10px;

    font-size: 16px;
    color: @search-box-border-color;
  }

  .search-box {
    padding-right: 30px;

    border-color: @search-box-border-color;

    &::placeholder {
      color: @search-box-placeholder-color;
      text-align: left;
    }

    &:hover {
      background: @search-box-background;
    }
  }

  .search-dropdown__icon {
    position: absolute;
    top: 0px;
    right: 2px;
    width: 30px;
    border-left: 1px solid @search-box-border-color;
    line-height: 26px;
    text-align: center;
    cursor: pointer;
    pointer-events: none;

    .sc-icon-caret-down {
      font-size: 7px;
      color: @search-box-border-color;
    }
  }
}
