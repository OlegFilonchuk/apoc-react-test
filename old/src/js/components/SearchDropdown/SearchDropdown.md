## Simple SearchDropdown

    const list = [{ label: 'first', value: 'fr' }, { label: 'second', value: 'sc' }, { label: 'third', value: 'th' }, { label: 'second first', value: 'sf' }, { label: 'third again', value: 'ta' }];

    const initialState = {
      list
    };

    function searchBoxChange(value) {
      const newList = list.filter(e => e.label.includes(value));

      setState({ list: newList });
    }

    function onOptionClick(value) {
      console.log(value, ' clicked');
    }

    (
      <SearchDropdown
        name="simple-example-dropdown"
        results={state.list}
        onSearchChange={searchBoxChange}
        onDropdownChange={onOptionClick}
      />
    )

## Simple SearchDropdown with empty results list

    const list = [];

    const initialState = {
      list
    };

    (
      <SearchDropdown name="empty-result-dropdown" results={state.list} />
    )

## Simple SearchDropdown with custom search icon

    const list = [];

    const initialState = {
      list
    };

    (
      <SearchDropdown name="empty-result-dropdown" results={state.list} searchIcon={<i className="sc-icon-down-arrow search-dropdown--icon " />} />
    )

## SearchDropdown with allowDynamicOptions

    const list = [{ label: 'first', value: 'fr' }, { label: 'second', value: 'sc' }, { label: 'third', value: 'th' }, { label: 'second first', value: 'sf' }, { label: 'third again', value: 'ta' }];

    const initialState = {
      list,
      searchValue: ''
    };

    function searchBoxChange(event) {
      setState({
        searchValue: event.target.value
      });
    }

    function onOptionClick({value, label}) {
      console.log(value, ' clicked');

      setState({
        list: [...state.list, { value, label }]
      });
    }

    (
      <SearchDropdown
        name="simple-example-dropdown"
        results={state.list}
        onSearchChange={searchBoxChange}
        onDropdownChange={onOptionClick}
        allowDynamicOptions
        searchValue={state.searchValue}
      />
    )
