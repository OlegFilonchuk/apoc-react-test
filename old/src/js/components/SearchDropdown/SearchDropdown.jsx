import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Dropdown from '../Dropdown/Dropdown';
import DropdownOption from '../Dropdown/DropdownOption';
import DropdownInput from '../Dropdown/DropdownInput';

import './SearchDropdown.less';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

class SearchDropdown extends React.Component {
  get filteredDropdownOptions() {
    const arrayOfOptionValues = Array.from(this.props.results)
      .filter(({ value }) => value && value.startsWith(this.props.searchValue))
      .includes(this.props.searchValue);

    const resultsArray = arrayOfOptionValues.length
      ? arrayOfOptionValues
      : [{ value: this.props.searchValue, label: this.props.searchValue }];

    return resultsArray.map(this.constructDropdownOption);
  }

  get dropdownOptions() {
    const { allowDynamicOptions, searchValue, results } = this.props;

    if (allowDynamicOptions) {
      return searchValue ? this.filteredDropdownOptions : Array.from(results).map(this.constructDropdownOption);
    }

    return results.length ? results.map(({ label, value }) => this.constructDropdownOption({ label, value })) : null;
  }

  get resultsDropdownButton() {
    const { placeholder, onSearchChange, allowDynamicOptions, searchValue } = this.props;

    return allowDynamicOptions ? (
      <input
        className="sc-btn search-box"
        onChange={onSearchChange}
        value={searchValue}
        {...getDataTestElementProps(this.props)}
        inputType={'searchDropdown'}
        placeholder={placeholder}
      />
    ) : (
      <DropdownInput
        className="sc-btn-default search-box"
        placeholder={placeholder}
        onChange={onSearchChange}
        {...getDataTestElementProps(this.props)}
      />
    );
  }

  get resultsDropdown() {
    const { name, onDropdownChange, useSubmitOnEnter } = this.props;

    return (
      <Dropdown
        name={name}
        onChange={onDropdownChange}
        useSubmitOnEnter={useSubmitOnEnter}
        button={this.resultsDropdownButton}
      >
        {this.dropdownOptions}
      </Dropdown>
    );
  }

  get searchIcon() {
    return this.props.allowDynamicOptions ? (
      <div className="search-dropdown__icon">
        <i className="sc-icon-caret-down" />
      </div>
    ) : (
      this.props.searchIcon
    );
  }

  get className() {
    return classNames('search-dropdown', this.props.className);
  }

  constructDropdownOption = ({ label, value }) => (
    <DropdownOption key={`${value}-${label}`} label={label} value={value} selected={value === this.props.searchValue} />
  );

  render() {
    return (
      <div className={this.className}>
        {this.resultsDropdown}
        {this.searchIcon}
      </div>
    );
  }
}

SearchDropdown.propTypes = {
  /**
   * Necessary only when [DropdownOption](#dropdownoption) is used. Sets a name every option's `input`.
   */
  name: PropTypes.string.isRequired,

  /**
   * An additional class name
   */
  className: PropTypes.string,

  /**
   * placeholder for dropdown input
   */
  placeholder: PropTypes.string,

  /**
   * onSearchChange handler. Process the search value.
   */
  onSearchChange: PropTypes.func,

  /**
   * results are values visible as dropdown options
   */
  results: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.string, label: PropTypes.string })),

  /**
   * onDropdownChange handler. Process the dropdown options.
   */
  onDropdownChange: PropTypes.func,

  /**
   * searchIcon is visible icon next to input search
   */
  searchIcon: PropTypes.element,

  /**
   * Enable/disable submit on enter press
   */
  useSubmitOnEnter: PropTypes.bool,

  /**
   * allowDynamicOptions - additional props which allows to add custom values to results (dropdown options)
   */
  allowDynamicOptions: PropTypes.bool,

  /**
   * searchValue - custom value which can be add to results (dropdown options)
   */
  searchValue: PropTypes.string,
  ...dataTestElementPropTypes
};

SearchDropdown.defaultProps = {
  className: '',
  placeholder: 'Search for country',
  onDropdownChange: () => true,
  onSearchChange: () => true,
  results: [],
  searchIcon: <i className="sc-icon-magnifier search-dropdown--icon" />,
  useSubmitOnEnter: false,
  allowDynamicOptions: false,
  searchValue: ''
};

export default SearchDropdown;
