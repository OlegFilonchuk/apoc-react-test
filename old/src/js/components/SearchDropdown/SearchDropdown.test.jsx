import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchDropdown from './SearchDropdown';
import Dropdown from '../Dropdown/Dropdown';
import DropdownInput from '../Dropdown/DropdownInput';

const defaultProps = { name: 'TestDropdown' };
const list = [
  { label: 'first', value: 'fr' },
  { label: 'second', value: 'sc' },
  { label: 'third', value: 'th' },
  { label: 'second first', value: 'sf' },
  { label: 'third again', value: 'ta' }
];

const mountSearchDropdown = (props = {}, mountMethod = mount) =>
  mountMethod(<SearchDropdown {...defaultProps} {...props} />);

describe('<SearchDropdown>', () => {
  it('should render <SearchDropdown>', () => {
    expect(toJson(mountSearchDropdown({ results: list }))).toMatchSnapshot();
  });

  it('should render <SearchDropdown> with empty options', () => {
    expect(toJson(mountSearchDropdown({ results: [] }))).toMatchSnapshot();
  });

  it('should open dropdown on input click', () => {
    const searchDropdown = mountSearchDropdown({ results: list });

    expect(searchDropdown.find(Dropdown).find('.sc-open').length).toEqual(0);

    searchDropdown
      .find(DropdownInput)
      .find('input')
      .simulate('click');

    expect(searchDropdown.find(Dropdown).find('.sc-open').length).toEqual(1);
  });

  it('should call onDropdownChange when the Enter button is pressed', () => {
    const onChange = jest.fn();
    const wrapper = mountSearchDropdown({ results: list, useSubmitOnEnter: true, onDropdownChange: onChange });

    expect(onChange).not.toHaveBeenCalled();

    wrapper
      .find(DropdownInput)
      .find('input')
      .simulate('click')
      .simulate('keyDown', { preventDefault: () => {}, keyCode: 13, key: 'Enter' }); // enter

    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
