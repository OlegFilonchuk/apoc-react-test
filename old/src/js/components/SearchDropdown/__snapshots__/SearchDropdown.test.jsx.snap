// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchDropdown> should render <SearchDropdown> 1`] = `
<SearchDropdown
  allowDynamicOptions={false}
  className=""
  name="TestDropdown"
  onDropdownChange={[Function]}
  onSearchChange={[Function]}
  placeholder="Search for country"
  results={
    Array [
      Object {
        "label": "first",
        "value": "fr",
      },
      Object {
        "label": "second",
        "value": "sc",
      },
      Object {
        "label": "third",
        "value": "th",
      },
      Object {
        "label": "second first",
        "value": "sf",
      },
      Object {
        "label": "third again",
        "value": "ta",
      },
    ]
  }
  searchIcon={
    <i
      className="sc-icon-magnifier search-dropdown--icon"
    />
  }
  searchValue=""
  useSubmitOnEnter={false}
>
  <div
    className="search-dropdown"
  >
    <Dropdown
      button={
        <DropdownInput
          className="sc-btn-default search-box"
          inputType="input"
          isExpanded={false}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder="Search for country"
          value=""
        />
      }
      className=""
      emptyMessage="No Filtered Data"
      filterable={false}
      fullWidth={false}
      name="TestDropdown"
      onChange={[Function]}
      recomputeUseButton={false}
      useSubmitOnEnter={false}
      withSuffix={Object {}}
    >
      <div
        className="sc-dropdown"
      >
        <DropdownInput
          className="sc-btn-default search-box"
          inputType="input"
          isExpanded={false}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder="Search for country"
          value=""
        >
          <input
            className="sc-btn sc-btn-default search-box"
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            placeholder="Search for country"
            type="input"
            value=""
          />
        </DropdownInput>
        <ul
          aria-labelledby={undefined}
          className="sc-dropdown-menus"
        >
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$fr-first"
            label="first"
            name="TestDropdown"
            onClick={[Function]}
            selected={true}
            showPopupOnHover={false}
            value="fr"
          >
            <li
              className="dropdown-option "
            >
              <label
                className="sc-focus"
                title={undefined}
              >
                <span
                  className=" icon"
                />
                first
                <input
                  data-test-element="option-fr"
                  defaultChecked={true}
                  disabled={false}
                  name="TestDropdown"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="fr"
                />
              </label>
            </li>
          </DropdownOption>
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$sc-second"
            label="second"
            name="TestDropdown"
            onClick={[Function]}
            selected={false}
            showPopupOnHover={false}
            value="sc"
          >
            <li
              className="dropdown-option "
            >
              <label
                className=""
                title={undefined}
              >
                <span
                  className=" icon"
                />
                second
                <input
                  data-test-element="option-sc"
                  defaultChecked={false}
                  disabled={false}
                  name="TestDropdown"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="sc"
                />
              </label>
            </li>
          </DropdownOption>
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$th-third"
            label="third"
            name="TestDropdown"
            onClick={[Function]}
            selected={false}
            showPopupOnHover={false}
            value="th"
          >
            <li
              className="dropdown-option "
            >
              <label
                className=""
                title={undefined}
              >
                <span
                  className=" icon"
                />
                third
                <input
                  data-test-element="option-th"
                  defaultChecked={false}
                  disabled={false}
                  name="TestDropdown"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="th"
                />
              </label>
            </li>
          </DropdownOption>
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$sf-second first"
            label="second first"
            name="TestDropdown"
            onClick={[Function]}
            selected={false}
            showPopupOnHover={false}
            value="sf"
          >
            <li
              className="dropdown-option "
            >
              <label
                className=""
                title={undefined}
              >
                <span
                  className=" icon"
                />
                second first
                <input
                  data-test-element="option-sf"
                  defaultChecked={false}
                  disabled={false}
                  name="TestDropdown"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="sf"
                />
              </label>
            </li>
          </DropdownOption>
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$ta-third again"
            label="third again"
            name="TestDropdown"
            onClick={[Function]}
            selected={false}
            showPopupOnHover={false}
            value="ta"
          >
            <li
              className="dropdown-option "
            >
              <label
                className=""
                title={undefined}
              >
                <span
                  className=" icon"
                />
                third again
                <input
                  data-test-element="option-ta"
                  defaultChecked={false}
                  disabled={false}
                  name="TestDropdown"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="ta"
                />
              </label>
            </li>
          </DropdownOption>
        </ul>
      </div>
    </Dropdown>
    <i
      className="sc-icon-magnifier search-dropdown--icon"
    />
  </div>
</SearchDropdown>
`;

exports[`<SearchDropdown> should render <SearchDropdown> with empty options 1`] = `
<SearchDropdown
  allowDynamicOptions={false}
  className=""
  name="TestDropdown"
  onDropdownChange={[Function]}
  onSearchChange={[Function]}
  placeholder="Search for country"
  results={Array []}
  searchIcon={
    <i
      className="sc-icon-magnifier search-dropdown--icon"
    />
  }
  searchValue=""
  useSubmitOnEnter={false}
>
  <div
    className="search-dropdown"
  >
    <Dropdown
      button={
        <DropdownInput
          className="sc-btn-default search-box"
          inputType="input"
          isExpanded={false}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder="Search for country"
          value=""
        />
      }
      className=""
      emptyMessage="No Filtered Data"
      filterable={false}
      fullWidth={false}
      name="TestDropdown"
      onChange={[Function]}
      recomputeUseButton={false}
      useSubmitOnEnter={false}
      withSuffix={Object {}}
    >
      <div
        className="sc-dropdown"
      >
        <DropdownInput
          className="sc-btn-default search-box"
          inputType="input"
          isExpanded={false}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder="Search for country"
          value=""
        >
          <input
            className="sc-btn sc-btn-default search-box"
            onChange={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            placeholder="Search for country"
            type="input"
            value=""
          />
        </DropdownInput>
        <ul
          aria-labelledby={undefined}
          className="sc-dropdown-menus"
        />
      </div>
    </Dropdown>
    <i
      className="sc-icon-magnifier search-dropdown--icon"
    />
  </div>
</SearchDropdown>
`;
